src/components/Modals/PurchaseModelVersionModal.tsx

```diff
@@ -0,0 +1,109 @@
+import { Badge, Button, CloseButton, Divider, Group, Stack, Text } from '@mantine/core';
+import React, { useEffect } from 'react';
+
+import { createContextModal } from '~/components/Modals/utils/createContextModal';
+import { useCurrentUser } from '~/hooks/useCurrentUser';
+import { showErrorNotification } from '~/utils/notifications';
+import { trpc } from '~/utils/trpc';
+import { UserBuzz } from '../User/UserBuzz';
+import { useModelVersionPurchase } from '~/components/Model/ModelVersions/useModelVersionPurchase';
+import { CurrencyBadge } from '~/components/Currency/CurrencyBadge';
+import { Currency } from '@prisma/client';
+
+const { openModal, Modal } = createContextModal<{
+  modelVersionId: number;
+  onSuccess?: () => void;
+}>({
+  name: 'purchaseModelVersion',
+  centered: true,
+  radius: 'lg',
+  withCloseButton: false,
+  Element: ({ context, props: { modelVersionId, onSuccess } }) => {
+    const currentUser = useCurrentUser();
+    const { price, isLoading } = useModelVersionPurchase({ modelVersionId });
+    const purchaseModelVersionMutation = trpc.modelVersionPurchase.purchase.useMutation({
+      async onSuccess() {
+        onSuccess?.();
+        context.close();
```

Comment:
Might be good to call `onClose` here in case there's more clean up happening in the future when closing the modal
```suggestion
        onClose();
```